### DISCUSSION TITLE
Hotkeys for Screenshot and Video

### DISCUSSION THREAD

**Discussion by Lulech23** [Valkirie/HandheldCompanion#261] [Ideas]
First off, I have to say I love the new hotkey manager! It offers options for _almost_ everything I want, save one (well, two): **capture controls**. On my handhelds, I always bind one key to press for screenshot, hold for video.

While different applications have different bindings for these functions, it seems logical to use the Windows defaults:

- Take Screenshot: Win + Alt + PrtScrn
- Start/Stop Recording: Win + Alt + R

If other applications are preferred, users could rebind the Windows keys to something else in GameBar settings and then assign their own applications to these bindings instead, so it's still a fairly universal solution.

If needed, I could make the change and do a PR myself, but it seems like a pretty straightforward addition. And it would really complete the experience!

**Comment by Valkirie** [comment]
Have a look at our latest release. I'm sure it'll suits your needs ;)
![image](https://user-images.githubusercontent.com/934757/195979124-560423f1-9fc0-4e73-aaca-c91e91f1c1c7.png)

https://github.com/Valkirie/ControllerService/releases/tag/0.12.1.2

**Reply by Lulech23** [reply]
Well, what perfect timing! Thanks for the update :)

**Reply by 5310** [reply]
Sorry for the ping on a resolved discussion, @Lulech23. But can you please share your Hotkeys config file for the screenshot custom shortcut? 

I am on a Steam Deck, and have no access to a physical keyboard atm to be able to configure it myself. I can edit the json files under `Documents/HandheldCompanion/hotkeys/...`, but I have no idea what the syntax is for Win+Alt+PrntScr. If you could share yours that would help a lot!

**Reply by Lulech23** [reply]
Unfortunately this feature has been broken for me for a while now. I can't even see anything under the "custom shortcuts" header, and my old bindings no longer work. I've been meaning to try a clean reinstall of HC, but haven't gotten around to it yet.

When custom shortcuts were working, though, I used Shift + F12 and then assigned that combo to both Steam and the Xbox overlay so that I'd get a capture regardless of what launcher I was in.

**Reply by 5310** [reply]
Bummer :(

I'd still appreciate it if you shared your json file though. There's just no way for me to assign shortcuts with the on screen keyboard for now, so editing the config is the only way...in case it works on my machine. The files should still be there, I believe!

**Reply by Lulech23** [reply]
So, I reinstalled HC and fixed my hotkeys. Here's the JSON for my shortcut for recording video through the XBox Game Bar (Win + Alt + R):

```
{
  "hotkeyId": 52,
  "inputsChord": {
    "GamepadButtons": 4294967296,
    "OutputKeys": [
      {
        "KeyValue": 91,
        "ScanCode": 91,
        "Timestamp": 521640,
        "IsKeyDown": true,
        "IsKeyUp": false,
        "IsExtendedKey": true
      },
      {
        "KeyValue": 164,
        "ScanCode": 56,
        "Timestamp": 521687,
        "IsKeyDown": true,
        "IsKeyUp": false,
        "IsExtendedKey": false
      },
      {
        "KeyValue": 82,
        "ScanCode": 19,
        "Timestamp": 522015,
        "IsKeyDown": true,
        "IsKeyUp": false,
        "IsExtendedKey": false
      },
      {
        "KeyValue": 82,
        "ScanCode": 19,
        "Timestamp": 522109,
        "IsKeyDown": false,
        "IsKeyUp": true,
        "IsExtendedKey": false
      },
      {
        "KeyValue": 164,
        "ScanCode": 56,
        "Timestamp": 522156,
        "IsKeyDown": false,
        "IsKeyUp": true,
        "IsExtendedKey": false
      },
      {
        "KeyValue": 91,
        "ScanCode": 91,
        "Timestamp": 522250,
        "IsKeyDown": false,
        "IsKeyUp": true,
        "IsExtendedKey": true
      }
    ],
    "InputsType": 2
  },
  "IsPinned": false,
  "Name": "Record Video"
}
```

I believe the keycode for Print Screen is `44` if you want to replace "R" (`82`).

**Reply by 5310** [reply]
(Shouldn't've replied to the mail, seeing as it created a new thread!)

Thank you _very_ much, again! 

I was doing it so wrong, just dumping the keycodes into the empty output lists! Added in the shortcuts and they're working great! :blush:
